### PR TITLE
Support endpoints in nACL generation

### PR DESCRIPTION
**Pull request by YairSlobodin1**

### CODE REVIEW

**Review by kyorav — COMMENTED** [review]
I tested a couple of scenarios, LGTM.

File: pkg/ir/unspecified.go [hunk 1]
```diff
@@ -50,7 +51,26 @@ func (s *Spec) ComputeBlockedSubnets() []ID {
...
+		for instanceName, instance := range s.Defs.Instances {
+			instanceInList := false
+			for _, nif := range instance.Nifs {
+				if subnet == s.Defs.NIFs[nif].Subnet {
+					nifs = append(nifs, nif)
+					if !instanceInList {
+						nifs = append(nifs, instanceName)
+						instanceInList = true
+					}
+					break
```
Comment (zivnevo) [inline pkg/ir/unspecified.go:69]: I don't understand this break - don't we want all NIFs in this list?

**Review by zivnevo — CHANGES_REQUESTED** [review]

File: pkg/synth/acl.go [hunk 2]
```diff
@@ -153,3 +156,12 @@ func subnetsContainedInCidr(s *ir.Spec, cidr *ipblock.IPBlock, resource ir.Resou
 	}
 	return result
 }
+
+func updateEndpoint(s *ir.ConfigDefs, resource ir.Resource, addr *ipblock.IPBlock) *ipblock.IPBlock {
```
[the remark this answers is not in the document]
Comment (YairSlobodin1) [reply pkg/synth/acl.go:160]: Should I change Src and Dst be `*Resource` [here](https://github.com/np-guard/vpc-network-config-synthesis/blob/main/pkg/ir/spec.go#L31)?

File: pkg/synth/acl.go [hunk 3]
```diff
@@ -153,3 +156,12 @@ func subnetsContainedInCidr(s *ir.Spec, cidr *ipblock.IPBlock, resource ir.Resou
...
+	if resource.Type == ir.ResourceTypeNIF {
+		nifName, _ := s.NIFFromIP(addr)
```
[the remark this answers is not in the document]
Comment (YairSlobodin1) [reply pkg/synth/acl.go:162]: we check this [before](https://github.com/np-guard/vpc-network-config-synthesis/blob/main/pkg/ir/spec.go#L297).

File: pkg/ir/unspecified.go [hunk 4]
```diff
@@ -42,13 +42,18 @@ func (s *Spec) ComputeBlockedSubnets() []ID {
 		// cidr segments which include the subnet
 		cidrSegments := []ID{}
 		for segmentName, cidrSegmentDetails := range s.Defs.CidrSegments {
+			subnetInSegment := false
```
Comment (zivnevo) [inline pkg/ir/unspecified.go:45]: Consider using [a labeled break statement](https://stackoverflow.com/questions/51996175/how-to-break-out-of-nested-loops-in-go) instead.
Another option is a labeled `continue` statement.

File: pkg/synth/acl.go (hunk 3 above)
[the remark this answers is not in the document]
Comment (zivnevo) [reply pkg/synth/acl.go:162]: got it. Maybe just add a comment

**Review by zivnevo — CHANGES_REQUESTED** [review]

**Review by zivnevo — APPROVED** [review]

**Pull request merged**